app/scripts/actions/index.js

```diff
@@ -303,6 +328,36 @@ export const applyWorkflowToGranule = (granuleId, workflow) => ({
   }
 });
 
+export const getCollectionByGranuleId = (dispatch, granuleId) => {
+  return dispatch(getGranule(granuleId)).then((granuleResponse) => {
+    const { name, version } = collectionNameVersion(granuleResponse.data.collectionId);
+    return dispatch(getCollection(name, version));
+  });
+};
+
+export const applyRecoveryWorkflowToGranule = (granuleId) => {
+  return (dispatch) => {
+    return getCollectionByGranuleId(dispatch, granuleId)
```

Comment:
Then this would be:

` return dispatch(getCollectionByGranuleId(granuleId))`

Reply:
nice catch :100: